fairlearn/metrics/_metric_frame.py

```diff
@@ -323,14 +330,107 @@ def __init__(
                 raise ValueError(_DUPLICATE_FEATURE_NAME.format(name))
             nameset.add(name)
 
-        # Create the 'overall' results
-        self._result = DisaggregatedResult.create(
+        # Create the basic results
+        result = DisaggregatedResult.create(
             data=all_data,
             annotated_functions=annotated_funcs,
             sensitive_feature_names=self._sf_names,
             control_feature_names=self._cf_names,
         )
 
+        # Build into cache
+        self._result_cache = dict()
+        self._populate_results(result)
+
+    def _populate_results(self, raw_result: DisaggregatedResult):
+        # Start with overall
```

Comment:
seems like each of these sections could be their own method/func

Reply:
I'm not convinced of the value of that, since the method isn't that long and will only be called once. However, I have refactored out a little of the common functionality, to make this method a bit shorter (and to avoid some almost-duplicated code).